extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/WrongAnnotationUsageProcessor.java

```diff
@@ -90,6 +91,16 @@ public String apply(AnnotationInstance annotationInstance) {
                             new IllegalStateException(String.format(
                                     "The %s class %s declares a producer but it must be ignored per the CDI rules",
                                     clazz.nestingType().toString(), clazz.name().toString()))));
+                } else {
+                    // detect interceptor bindings on nested classes
+                    for (DotName interceptorBinding : interceptorResolverBuildItem.getInterceptorBindings()) {
```

Comment:
We should rather use `io.quarkus.arc.processor.Annotations.containsAny(Collection<AnnotationInstance>, Iterable<DotName>)` with the transformed annotations (i.e. `classAnnotations`) as the first argument and `interceptorResolverBuildItem.getInterceptorBindings()` as the second argument.

Reply:
Using `classAnnotations` isn't enough as those are only for class-level annotation and we need method-level as well. I have combined the two approaches.